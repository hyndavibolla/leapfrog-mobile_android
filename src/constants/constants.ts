export const pushNotificationTypes = {
  USER_INACTIVE: 'USER_INACTIVE',
  MISSION_COMPLETE: 'MISSION_COMPLETE',
  BALANCE_NEW_POINTS: 'BALANCE_NEW_POINTS',
  BALANCE_EXPIRING_POINTS: 'BALANCE_EXPIRING_POINTS',
  BALANCE_UNUSED_POINTS: 'BALANCE_UNUSED_POINTS',
  MARKETING: 'MARKETING',
  USER_BIRTHDAY: 'USER_BIRTHDAY'
};

export const RedemptionMethods = {
  IN_STORE_ONLY: 'In-Store only',
  ONLINE_ONLY: 'Online only',
  IN_STORE_OR_ONLINE_ONLY: 'In-Store and Online only',
  IN_RESTAURANT_ONLY: 'In Restaurant only',
  APP_ONLY: 'App only'
};

export const GROCERY_AND_DELIVERY_CATEGORY = 'Grocery & Delivery';

export const TutorialFrom = {
  TUTORIAL_REMINDER: 'tutorial_reminder',
  TUTORIAL_BANNER: 'tutorial_banner',
  PROFILE: 'profile'
};
