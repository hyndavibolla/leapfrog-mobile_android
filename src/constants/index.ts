export * from './env';
export * from './routes';
export * from './eventTracking';
export * from './styles';
export * from './icons';
export * from './constants';
