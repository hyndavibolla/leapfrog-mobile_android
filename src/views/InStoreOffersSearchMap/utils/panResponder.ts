const swipeConfig = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
  gestureIsClickThreshold: 5
};

const swipeDirections = {
  swipeUp: 'SWIPE_UP',
  swipeDown: 'SWIPE_DOWN'
};

export const handleShouldSetPanResponder = ({ nativeEvent: { touches } }, gestureState) => {
  return touches.length === 1 && !gestureIsClick(gestureState);
};

export const gestureIsClick = ({ dx, dy }) => {
  return Math.abs(dx) < swipeConfig.gestureIsClickThreshold && Math.abs(dy) < swipeConfig.gestureIsClickThreshold;
};

export const handlePanResponderEnd = (gestureState, onSwipeUp, onSwipeDown) => {
  const swipeDirection = getSwipeDirection(gestureState);
  triggerSwipeHandlers(swipeDirection, onSwipeUp, onSwipeDown);
};

export const triggerSwipeHandlers = (swipeDirection, onSwipeUp, onSwipeDown) => {
  const { swipeUp, swipeDown } = swipeDirections;
  switch (swipeDirection) {
    case swipeUp:
      onSwipeUp();
      break;
    case swipeDown:
      onSwipeDown();
      break;
  }
};

export const getSwipeDirection = gestureState => {
  const { swipeUp, swipeDown } = swipeDirections;
  if (isValidVerticalSwipe(gestureState)) {
    return gestureState.dy > 0 ? swipeDown : swipeUp;
  }
  return null;
};

export const isValidVerticalSwipe = ({ vy, dx }) => {
  const { velocityThreshold, directionalOffsetThreshold } = swipeConfig;
  return isValidSwipe(vy, velocityThreshold, dx, directionalOffsetThreshold);
};

export const isValidSwipe = (velocity, velocityThreshold, directionalOffset, directionalOffsetThreshold) => {
  return Math.abs(velocity) > velocityThreshold && Math.abs(directionalOffset) < directionalOffsetThreshold;
};
