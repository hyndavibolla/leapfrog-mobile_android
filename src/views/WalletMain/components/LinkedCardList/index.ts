export { default as LinkedCardList } from './LinkedCardList';
