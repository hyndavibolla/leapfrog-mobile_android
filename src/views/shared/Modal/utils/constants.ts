export enum ModalType {
  CENTER = 'center',
  BOTTOM = 'bottom'
}

export enum ModalSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  EXTRA_LARGE = 'extralarge',
  DYNAMIC = 'dynamic',
  FULL_SCREEN = 'fullscreen'
}

export const ModalSizeToHeightMap = {
  [ModalSize.SMALL]: 40,
  [ModalSize.MEDIUM]: 50,
  [ModalSize.EXTRA_LARGE]: 80,
  [ModalSize.FULL_SCREEN]: 100
};
