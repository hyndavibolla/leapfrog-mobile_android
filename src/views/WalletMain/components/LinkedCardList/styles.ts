import { css } from '@emotion/native';
import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../../../constants';

export const styles = {
  banner: css`
    background: ${COLOR.WHITE};
    border-radius: 8px;
    padding: 20px 16px;
    margin-bottom: 8px;
    flex-direction: row;
  `,
  bannerContent: css`
    flex: 1;
    margin-left: 20px;
  `,
  bannerTitle: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
    font-family: ${FONT_FAMILY.HEAVY};
  `,
  bannerDescription: css`
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.PETITE};
    line-height: ${FONT_SIZE.REGULAR};
    font-family: ${FONT_FAMILY.MEDIUM};
    margin-bottom: 4px;
  `,
  bannerActionsContainer: css`
    flex-direction: row;
    align-items: center;
  `,
  bannerTextAction: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-size: ${FONT_SIZE.PETITE};
    line-height: ${FONT_SIZE.SMALL};
    font-family: ${FONT_FAMILY.HEAVY};
    margin-right: 8px;
  `,
  footNote: css`
    color: ${COLOR.DARK_GRAY};
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.TINY};
    line-height: ${FONT_SIZE.SMALL};
    margin-top: 8px;
    text-align: right;
  `,
  cardFootNote: css`
    margin-top: 4px;
    text-align: left;
  `,
  loading: css`
    margin-top: 48px;
  `,
  image: css`
    height: 30px;
    width: 50px;
  `,
  addAnotherCardContainer: css`
    align-items: flex-end;
    margin-bottom: 28px;
  `,
  addAnotherCardText: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.SMALL};
    font-family: ${FONT_FAMILY.BOLD};
  `
};
