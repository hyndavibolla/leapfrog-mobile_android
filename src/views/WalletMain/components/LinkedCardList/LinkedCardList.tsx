import React, { memo, useCallback } from 'react';
import { Text, View, Pressable, TouchableHighlight } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { EmptyState } from '../../../shared/EmptyState';

import { CONTAINER_STYLE, ICON, ROUTES, COLOR } from '_constants';
import { useTestingHelper } from '_utils/useTestingHelper';
import CreditCardIcon from '_assets/in-store-offer/creditCard.svg';
import CardLinkFallbackIcon from '_assets/in-store-offer/cardLinkFallback.svg';
import { styles } from './styles';
import { ILinkedCard } from '_models/cardLink';
import { LinkedCardListItem } from '../LinkedCardListItem';
import { Icon } from '_commons/components/atoms/Icon';

export interface Props {
  linkedCardsListError: any;
  linkedCardsList: ILinkedCard[];
}

export const LinkedCardList = ({ linkedCardsListError, linkedCardsList }: Props) => {
  const { getTestIdProps } = useTestingHelper('linked-cards');
  const { navigate } = useNavigation();

  const handlePressAddCardButton = useCallback(() => {
    navigate(ROUTES.IN_STORE_OFFERS.CARD_LINK);
  }, [navigate]);

  if (linkedCardsListError) {
    return (
      <EmptyState visible Icon={CardLinkFallbackIcon} title="Whoops! We had a problem loading your linked cards." card {...getTestIdProps('empty-state')} />
    );
  }
  return (
    <>
      {linkedCardsList.map(({ cardId, cardType, cardLastFour, isSywCard }) => (
        <LinkedCardListItem key={cardId} cardId={cardId} cardType={cardType} cardLastFour={cardLastFour} isSywCard={isSywCard} isLinkedToCardlink />
      ))}
      {!linkedCardsList?.length && (
        <TouchableHighlight underlayColor="transparent" onPress={handlePressAddCardButton} {...getTestIdProps('container')}>
          <View style={[CONTAINER_STYLE.shadow, styles.banner]}>
            <CreditCardIcon width={40} height={40} />
            <View style={styles.bannerContent}>
              <Text style={styles.bannerTitle}>Eat and Earn</Text>
              <Text style={styles.bannerDescription}>
                Add any Visa, Mastercard, or{'\n'}American Express to earn points{'\n'}near you
              </Text>
              <View style={styles.bannerActionsContainer} {...getTestIdProps('add-card-btn')}>
                <Text style={styles.bannerTextAction}>Add my card</Text>
                <Icon name={ICON.ARROW_RIGHT} color={COLOR.BLACK} />
              </View>
            </View>
          </View>
        </TouchableHighlight>
      )}
      {linkedCardsList?.length ? (
        <Pressable style={styles.addAnotherCardContainer} onPress={handlePressAddCardButton} {...getTestIdProps('add-another-card-btn')}>
          <Text style={styles.addAnotherCardText}>+ Add Card</Text>
        </Pressable>
      ) : null}
    </>
  );
};
export default memo(LinkedCardList);
