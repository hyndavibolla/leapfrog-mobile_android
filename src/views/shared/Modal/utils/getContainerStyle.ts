import { memorize } from '../../../../utils/memorize';
import { styles } from '../styles';
import { ModalSize, ModalType, ModalSizeToHeightMap } from './constants';

interface props {
  height: number;
  modalHeight: number;
  size: ModalSize;
  type: ModalType;
}

const getContainerStyle = ({ height, modalHeight, size, type }: props) => {
  switch (type) {
    case ModalType.BOTTOM:
      if (size === ModalSize.FULL_SCREEN) return [styles.containerBottom, styles.containerBottomFullScreen];
      else if (size !== ModalSize.DYNAMIC) return [styles.containerBottom, { maxHeight: (ModalSizeToHeightMap[size] * height) / 100 }];
      return [styles.containerBottom];
    case ModalType.CENTER:
      if (size === ModalSize.DYNAMIC) return [styles.containerCenter, styles.containerCenterDynamic, { bottom: (height - modalHeight) / 2 }];
      const center = (ModalSizeToHeightMap[size] * height) / 100;
      return [styles.containerCenter, { height: center, bottom: (height - center) / 2 }];
  }
};

export default memorize(getContainerStyle);
