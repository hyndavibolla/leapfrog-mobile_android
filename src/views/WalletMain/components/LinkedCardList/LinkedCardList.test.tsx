import React from 'react';
import LinkedCardList, { Props } from './LinkedCardList';
import { renderWithGlobalContext } from '../../../../test-utils/renderWithGlobalContext';
import { getMockDeps } from '../../../../test-utils/getMockDeps';
import { Deps } from '../../../../models/general';
import { act } from 'react-test-renderer';
import { wait } from '../../../../utils/wait';
import { getLinkedCards_1, getLinkedCards_2 } from '../../../../test-utils/entities';
import { fireEvent } from '@testing-library/react-native';
import { ROUTES } from '../../../../constants';
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({ useNavigation: () => ({ navigate: mockNavigate }), useIsFocused: () => true }));
describe('LinkedCardList', () => {
  let deps: Deps;
  let props: Props;
  beforeEach(() => {
    deps = getMockDeps();
    props = {
      linkedCardsListError: null,
      linkedCardsList: []
    };
  });
  it('should render', async () => {
    const { toJSON } = renderWithGlobalContext(<LinkedCardList {...props} />, deps);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render empty container banner', async () => {
    props.linkedCardsListError = 'error';
    const { queryByTestId, toJSON } = renderWithGlobalContext(<LinkedCardList {...props} />, deps);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
    expect(queryByTestId('empty-state-card')).toBeTruthy();
  });
  it('should navigate to add card webview', async () => {
    props.linkedCardsList = [getLinkedCards_2(), getLinkedCards_1()];
    const { queryAllByTestId } = renderWithGlobalContext(<LinkedCardList {...props} />, deps);
    await act(() => wait(0));
    fireEvent.press(queryAllByTestId('linked-cards-add-another-card-btn')[0]);
    expect(mockNavigate).toBeCalledWith(ROUTES.IN_STORE_OFFERS.CARD_LINK);
  });
});
