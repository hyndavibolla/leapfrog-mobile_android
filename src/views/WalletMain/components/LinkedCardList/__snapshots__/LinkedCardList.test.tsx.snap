// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedCardList should render 1`] = `
<View
  accessibilityLabel="linked-cards-container"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="linked-cards-container"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "row",
          "marginBottom": 8,
          "paddingBottom": 20,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 20,
        },
      ]
    }
  >
    CreditCardIcon
    <View
      style={
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Heavy",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      >
        Eat and Earn
      </Text>
      <Text
        style={
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Medium",
            "fontSize": 14,
            "lineHeight": 20,
            "marginBottom": 4,
          }
        }
      >
        Add any Visa, Mastercard, or
        

        American Express to earn points
        

        near you
      </Text>
      <View
        accessibilityLabel="linked-cards-add-card-btn"
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
        testID="linked-cards-add-card-btn"
      >
        <Text
          style={
            Object {
              "color": "#0066CC",
              "fontFamily": "SFProDisplay-Heavy",
              "fontSize": 14,
              "lineHeight": 18,
              "marginRight": 8,
            }
          }
        >
          Add my card
        </Text>
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#262C2F",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          î¤ƒ
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`LinkedCardList should render empty container banner 1`] = `
<View
  accessibilityLabel="empty-state-card"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBEDF2",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "flexDirection": "row",
      "height": 125,
      "justifyContent": "space-between",
      "marginTop": 16,
      "paddingBottom": 28,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 28,
    }
  }
  testID="empty-state-card"
>
  CardLinkFallbackIcon
  <View
    style={
      Object {
        "paddingLeft": 20,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 20,
            "lineHeight": 30,
          },
          Object {
            "color": "#262C2F",
            "fontSize": 16,
            "fontWeight": "bold",
            "lineHeight": 24,
          },
        ]
      }
      type="titleHeader"
    >
      Whoops! We had a problem loading your linked cards.
    </Text>
  </View>
</View>
`;
